tasks/abs.rb

```diff
@@ -87,8 +87,18 @@ def tear_down(node_name, inventory_location)
 action = params['action']
 node_name = params['node_name']
 inventory_location = params['inventory']
-raise 'specify a node_name if tearing down' if action == 'tear_down' && node_name.nil?
-raise 'specify a platform if provisioning' if action == 'provision' && platform.nil?
+raise 'specify a node_name when tearing down' if action == 'tear_down' && node_name.nil?
+raise 'specify a platform when provisioning' if action == 'provision' && platform.nil?
```

Comment:
This would already raise an error if the parameter didn't match the action. Wouldn't it make more sense to move these checks down when doing `case action` ?

Reply:
No, because the there's three scenarios in which we want to raise an error:

1. When the user is provisioning a machine and forgot to specify a platform
2. When the user is tearing down a machine and forgot to specify a node name
3. When the user is performing any action and specified both node name and platform

Number three actually has 3 distinct error messages, depending on condition:

1. When both are specified and only node name should be (tearing down)
2. When both are specified and only platform should be (provisioning)
3. When both are specified and only one should be (any other action)